cmd/generateproxy/service.go

```diff
@@ -0,0 +1,133 @@
+// The MIT License
+//
+// Copyright (c) 2022 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"go/types"
+	"os"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+	"golang.org/x/tools/go/types/typeutil"
+
+	_ "go.temporal.io/api/workflowservice/v1"
+)
+
+const serviceFile = "../../proxy/service.go"
+
+const ServiceHeader = Header + `
+package proxy
+
+import (
+	"context"
+
+	"go.temporal.io/api/workflowservice/v1"
+)
+
+// WorkflowServiceProxyOptions provides options for configuring a WorkflowServiceProxyServer.
+// Client is a WorkflowServiceClient used to forward requests received by the server to the 
+// Temporal Frontend.
+type WorkflowServiceProxyOptions struct {
+	Client workflowservice.WorkflowServiceClient
+}
+
+type workflowServiceProxyServer struct {
+	workflowservice.UnimplementedWorkflowServiceServer
+	client workflowservice.WorkflowServiceClient
+}
+
+// NewWorkflowServiceProxyServer creates a WorkflowServiceServer suitable for registering with a GRPC Server. Requests will
```

Comment:
```suggestion
// NewWorkflowServiceProxyServer creates a WorkflowServiceServer suitable for registering with a gRPC Server. Requests will
```

please, everywhere...